types/index.test-d.ts

```diff
@@ -102,3 +107,72 @@ expectType<StyleDictionary.Core>(
     transforms: ["attribute/cti", "size/pt", "name/cti"],
   })
 );
+
+expectType<StyleDictionary.Core>(
+  StyleDictionary.registerParser({
+    pattern: /\.json$/,
+    parse: ({ contents, filePath }) => {
+      return {}
+    }
+  })
+);
+
+const file: StyleDictionary.File = {
+  destination: `somePath.json`,
+  format: `css/variables`,
+  filter: (token) => {
+    expectType<StyleDictionary.TransformedToken>(token);
+    return false;
+  },
+  options: {
+    showFileHeader: true,
+  }
+}
+
+expectType<StyleDictionary.Options | undefined>(file.options);
+
+// Files need a destination
+expectError<StyleDictionary.File>({
+  format: `css/variables`,
+});
+
+expectType<StyleDictionary.File>({
+  format: `css/variables`,
+  destination: `variables.css`
+} as StyleDictionary.File);
```

Comment:
I missed this while first reviewing, the `as` keyword in typescript is a [type assertion](https://www.typescriptlang.org/docs/handbook/basic-types.html#type-assertions).  So here you are saying to typescript to force the type of that object into `StyleDictionary.File`, which means even if the object did not match that type, this would pass.

While TypeScript only allows type assertions which convert to a more specific or less specific version of a type ("impossible conversions" such as `1 as StyleDictionary.File`), it will not prevent transforms such as `{format: 'css/variables'} as StyleDictionary.File` (File is typed to always require a destination). 

An alternative here would be to use the `expectAssignable` from tsd, which will ensure that this object can be assigned to `StyleDictionary.File`